app/components/ui/files/publish-modal/component.js

```diff
@@ -17,18 +17,41 @@ export default Ember.Component.extend({
     // An array that holds a pair, (fileName, fileSize)
     fileList: [],
     // Holds an array of objects that the user cannot be exclude from their package
-    nonOptionalFile: ['tale.yaml', 'environment.zip'],
+    nonOptionalFile: ['tale.yaml', 'environment.tar', 'license.txt'],
     // A map that connects the repository dropdown to a url
     repositoryMapping: {'Development': 'https://dev.nceas.ucsb.edu/knb/d1/mn/'},
     // The url for the published tale. This is set after publication succeeds
     packageUrl: '',
     // The jwt token that allows the user to interact with DataONE
     dataoneJWT: '',
+    // The name of the tale
+    taleName: '',
+
+    didInsertElement() {
+        this._super(...arguments);
+
+        this.getTaleFiles();
+        this.setTaleName();
+    },
+
+    didRender () {
+        // Create the tooltips after the template has been rendered
+        this.create_tooltips();
+    },
 
     setPublishBtnState(state) {
         this.set('enablePublish', state);
     },
 
+    setTaleName() {
+        let url = config.apiUrl + '/tale/'+ this.get('modalContext') 
```

Comment:
I thought I commented on this yesterday but can't seem to find it...

Is there a more Ember-y way to do this? I see two things that struck me as odd: (1) the URL for the request is constructed manually and this results in the user's browser sending an extra request for the Tale information when it already has it, in theory.

For (1), I'd expect something more concise like:

```js
self.set('taleName', new Tale(this.get('modalContext').get('title'))
```

For (2), does Ember let you give the modal the Tale information when it's built or can it peek outside itself, effectively sharing the information that was retrieved outside the modal's lifecycle?

Reply:
Oh, I did make this comment. Found it! https://github.com/whole-tale/dashboard/pull/222 Nvmd.

Reply:
I'll rebase/merge this branch with the parent with my next commit if that helps